cmd/charm/charmcmd/cmd.go

```diff
@@ -180,17 +178,9 @@ func newCharmStoreClient(ctxt *cmd.Context, auth authInfo, channel params.Channe
 		if err := agent.SetUpAuth(bakeryClient, agentInfo); err != nil {
 			return nil, errgo.Notef(err, "cannot set up agent authentication")
 		}
-	} else if auth.noBrowser {
-		tokenStore := ussologin.NewFileTokenStore(ussoTokenPath())
-		bakeryClient.AddInteractor(ussologin.NewInteractor(ussologin.StoreTokenGetter{
-			Store: tokenStore,
-			TokenGetter: ussologin.FormTokenGetter{
-				Filler: filler,
-				Name:   "charm",
-			},
-		}))
+	} else if !auth.noBrowser {
```

Comment:
you should probably also remove 	
in addAuthFlags
f.BoolVar(&info.noBrowser, "B", false, "do not use web browser for authentication")
f.BoolVar(&info.noBrowser, "no-browser-login", false, "")
and the noBrowser in authInfo strcut